eloquentjs/6_3_groups_iterable.js

```diff
@@ -0,0 +1,104 @@
+class GroupIterator {
+    [Symbol.iterator](set) {
+        this.set = set;
+        return this;
+    }
+
+    next() {
+        if(isNaN(this.n)){
```

Comment:
Instead of defining n in the next method I recommend defining it in the constructor. It will keep the logic simpler.
Also, add a space after the if.